addons/package_hierarchy/tests/test_package_hierarchy.py

```diff
@@ -0,0 +1,839 @@
+"""Test odoo-package-hierarchy"""
+
+from odoo.exceptions import ValidationError
+
+from . import common
+
+# Note that quant actually is being used; action_assign finds it.
+class TestPackageHierarchy(common.BaseHierarchy):
+    """Tests for odoo-package-hierarchy.
+
+    These tests are oriented for coverage, but do some correctness checks."""
+    def setUp(self):
+        """Set up common prerequisites for testing odoo-package-hierarchy.
```

Comment:
nit: There is no consistency on the comments, some of them start with space and some don't.